projects/plugins/jetpack/extensions/blocks/videopress/edit.js

```diff
@@ -596,40 +601,84 @@ const VideoPressEdit = CoreVideoEdit =>
+					this.props.setAttributes( { src: item.url } );
+				}
+			};
+
+			const uploadFinished = ( videoGuid = null ) => {
+				this.setState( { fileForUpload: null } );
+				if ( videoGuid ) {
+					setAttributes( { guid: videoGuid } );
+				}
+			};
+
+			const isResumableUploading = null !== fileForUpload && fileForUpload instanceof File;
+			if ( isResumableUploading ) {
+				return (
+					<VideoPressBlockProvider onUploadFinished={ uploadFinished }>
+						<Fragment>
+							{ blockSettings }
+							<ResumableUpload file={ fileForUpload } { ...this.props } />
+						</Fragment>
+					</VideoPressBlockProvider>
+				);
+			}
+
 			/*
 			 * The Loading/CoreVideoEdit blocks should be in the tree if :
 			 *     - We don't have a video GUID
 			 *     - Or we're uploading a video
 			 *     - Or we're in fallback mode (to display a video hosted locally for instance)
 			 * In all other cases, we should be able to safely display the Loading/VpBlock branch.
 			 */
+
 			const isFetchingVideo = isFetchingMedia || isFetchingPreview;
 			const renderCoreVideoAndLoadingBlocks = fallback || isUploading || ! guid;
 			const displayCoreVideoBlock =
 				renderCoreVideoAndLoadingBlocks && ! isUploading && ! isFetchingVideo;
 
 			// In order for the media placeholder to keep its state for error messages, we need to keep the CoreVideoEdit component in the tree during file uploads.
 			// Keep this section separate so the CoreVideoEdit stays in the tree, once we have a video, we don't need it anymore.
+			const coreVideoFragment = (
+				<Fragment>
+					<div className={ ! isUploading && ! isFetchingVideo ? 'videopress-block-hide' : '' }>
+						<Loading
+							text={
+								isUploading
+									? __( 'Uploading…', 'jetpack' )
+									: __(
+											'Generating preview…',
+											'jetpack',
+											/* dummy arg to avoid bad minification */ 0
+									  )
+							}
+						/>
+					</div>
+					<div className={ ! displayCoreVideoBlock ? 'videopress-block-hide' : '' }>
+						<CoreVideoEdit { ...this.props } />
+					</div>
+				</Fragment>
+			);
+
 			if ( renderCoreVideoAndLoadingBlocks ) {
-				return (
-					<Fragment>
-						<div className={ ! isUploading && ! isFetchingVideo ? 'videopress-block-hide' : '' }>
-							<Loading
-								text={
-									isUploading
-										? __( 'Uploading…', 'jetpack' )
-										: __(
-												'Generating preview…',
-												'jetpack',
-												/* dummy arg to avoid bad minification */ 0
-										  )
-								}
-							/>
-						</div>
-						<div className={ ! displayCoreVideoBlock ? 'videopress-block-hide' : '' }>
-							<CoreVideoEdit { ...this.props } />
-						</div>
-					</Fragment>
+				return resumableUploadEnabled ? (
+					<VideoPressBlockProvider
+						onFilesSelected={ filesSelected }
+						onMediaItemSelected={ mediaItemSelected }
+					>
+						{ coreVideoFragment }
+					</VideoPressBlockProvider>
+				) : (
+					<Fragment>{ coreVideoFragment }</Fragment>
```

Comment:
Maybe I don't understand the reason, but it seems to me we could return coreVideoFragment instead of having to encapsulate it in another Fragment ?

Reply:
I think it requires a wrapper in order to function properly. It doesn't have to be a `<Fragment>` though, I could do:

`<>{ coreVideoFragment }</>`

Or I could have the `coreVideoFragment` use the `<></>` and have the other two places where it is used wrap it in a `<Fragment>`?

Reply:
I thought we could directly include the coreVideoFragment, but if not, let's leave it like that :)